### DISCUSSION TITLE
[Feature Request]: Add Folders to SQL Editor

### DISCUSSION THREAD

**Discussion by raideno** [supabase/supabase#21177] [Feature Requests]
Hey i hope everyone reading this is good, First of all i want to thank the supabase team and all the contributors for this amazing product.

I use a lot the SQL Editor inside of supabase and it have been improved during the last updates but it still lacks some features.

My sugestion is to add a folders / directories system to the sql-editor in order to group similar queries together into groups / folders. This will also make it possible to have a sort option that sort queries alphabetically or by group.
A group could have a name, description and a color maybe.

In some of my projects i have hundreds of queries and some time it's really hard to find one of my queries even by naming it correctly.
![image](https://github.com/supabase/supabase/assets/67718820/226ee375-047e-4e27-b77c-258cfd64e53b)

Thanks.

**Comment by saltcod** [comment]
Definitely noted, thanks @raideno!

**Comment by joewinke** [comment]
Um, I had folders and then this morning they disappeared. Was I in some beta or something? I came looking to see what happened and then this post indicates they haven't been done, but yet I had it, and now its gone? @saltcod does this make sense?

**Reply by joewinke** [reply]
OK this is strange. I open up again and now my folder 'Tables & Views' is back. I found a tab that was previously open and then thought I would record the page refresh, and voila! it is back. 
![f129fb80b2fcb3d2dac0fc29f9b2ed71](https://github.com/user-attachments/assets/b2d5127a-5673-4c73-88b6-564d9cce7997)

**Comment by jrtnq514** [comment]
I have folders for my private queries but it would be awesome to have them in shared queries as well! 🤞

**Reply by toddmetheny** [reply]
Really, really want this as well.

**Reply by memke1** [reply]
Really want this, too!

**Reply by josh-weimer** [reply]
Yes! For my org this would be super helpful
